pkg/volume/util/nestedpendingoperations/nestedpendingoperations_test.go

```diff
@@ -590,129 +593,90 @@ func Test_NestedPendingOperations_Positive_WaitWithExpBackoff(t *testing.T) {
-		v1.UniqueVolumeName("volume-name"),
-		EmptyUniquePodName,
-		EmptyNodeName)
-}
+func Test_NestedPendingOperations_SecondOpBeforeFirstCompletes(t *testing.T) {
+	const (
+		keyNone = iota
+		keyVolume
+		keyVolumePod
+		keyVolumeNode
+	)
 
-func Test_NestedPendingOperations_Positive_SecondOpBeforeFirstCompletes_FirstVolumeNode_SecondNone(t *testing.T) {
-	testConcurrentOperationsPositive(t,
-		v1.UniqueVolumeName("volume-name"),
-		EmptyUniquePodName,
-		types.NodeName("operation-nodename"),
-		EmptyUniqueVolumeName,
-		EmptyUniquePodName,
-		EmptyNodeName)
-}
+	type testCase struct {
+		testId     int
+		keyTypes   []int // only 2 elements are supported
+		expectPass bool
+	}
+
+	tests := []testCase{
+		{testId: 1, keyTypes: []int{keyNone, keyNone}, expectPass: true},
+		{testId: 2, keyTypes: []int{keyNone, keyVolume}, expectPass: true},
+		{testId: 3, keyTypes: []int{keyNone, keyVolumePod}, expectPass: true},
+		{testId: 4, keyTypes: []int{keyNone, keyVolumeNode}, expectPass: true},
+		{testId: 5, keyTypes: []int{keyVolume, keyNone}, expectPass: true},
+		{testId: 6, keyTypes: []int{keyVolume, keyVolumePod}, expectPass: false},
+		{testId: 7, keyTypes: []int{keyVolume, keyVolumeNode}, expectPass: false},
+		{testId: 8, keyTypes: []int{keyVolumePod, keyNone}, expectPass: true},
+		{testId: 9, keyTypes: []int{keyVolumePod, keyVolume}, expectPass: false},
+		{testId: 10, keyTypes: []int{keyVolumeNode, keyNone}, expectPass: true},
+		{testId: 11, keyTypes: []int{keyVolumeNode, keyVolume}, expectPass: false},
+		{testId: 12, keyTypes: []int{keyVolumeNode, keyVolumeNode}, expectPass: false},
+	}
+
+	for _, test := range tests {
+		var (
+			volumeNames []v1.UniqueVolumeName
+			podNames    []volumetypes.UniquePodName
+			nodeNames   []types.NodeName
+		)
+		for _, keyType := range test.keyTypes {
+			var (
+				v v1.UniqueVolumeName
+				p volumetypes.UniquePodName
+				n types.NodeName
+			)
+			switch keyType {
+			case keyNone:
+				v = EmptyUniqueVolumeName
+				p = EmptyUniquePodName
+				n = EmptyNodeName
+			case keyVolume:
+				v = v1.UniqueVolumeName("volume-name")
+				p = EmptyUniquePodName
+				n = EmptyNodeName
+			case keyVolumePod:
+				v = v1.UniqueVolumeName("volume-name")
+				p = volumetypes.UniquePodName("operation-podname")
+				n = EmptyNodeName
+			case keyVolumeNode:
+				v = v1.UniqueVolumeName("volume-name")
+				p = EmptyUniquePodName
+				n = types.NodeName("operation-nodename")
+			}
+			volumeNames = append(volumeNames, v)
+			podNames = append(podNames, p)
+			nodeNames = append(nodeNames, n)
+		}
 
-func Test_NestedPendingOperations_Negative_SecondOpBeforeFirstCompletes_FirstVolumeNode_SecondVolume(t *testing.T) {
-	testConcurrentOperationsNegative(t,
-		v1.UniqueVolumeName("volume-name"),
-		EmptyUniquePodName,
-		types.NodeName("operation-nodename"),
-		v1.UniqueVolumeName("volume-name"),
-		EmptyUniquePodName,
-		EmptyNodeName)
-}
+		if test.expectPass {
+			testConcurrentOperationsPositive(t, test.testId,
```

Comment:
It is better to use `t.Run()` than to pass the testId into the helper function. e.g.

```
t.Run(fmt.Sprintf("test %d", test.testId), func(t *testing.T){
    if test.expectPass...
})